### PR TITLE
Check and lint sphinx configuration file doc/conf.py

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Run flake8 and pylint to check and lint doc/conf.py.

It would be easier to review each commit separately.

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -1,11 +1,8 @@
 # -*- coding: utf-8 -*-
-import sys
-import os
-import glob
-import shutil
+# pylint: disable=invalid-name,redefined-builtin,redefined-builtin
+# pylint: disable=missing-module-docstring
```
Comment (weiji14) [inline doc/conf.py:3]:
```suggestion
# -*- coding: utf-8 -*-
"""
Sphinx documentation configuration file.
"""
# pylint: disable=invalid-name
```

Also, if you can change this line:

https://github.com/GenericMappingTools/pygmt/blob/b2a1758555ce94465a29ab7318ca944c4074833c/doc/conf.py#L99

to use f-strings, and disable the `redefined-builtin` pylint warning:

```python
copyright = f"2017-{year}, The PyGMT Developers."  # pylint: disable=redefined-builtin
```

Comment (seisman) [reply doc/conf.py]: Fixed in d94dbe7eb8beee1ccb14be3216dbc18c05f3800b.

**Comment by weiji14** [comment]
Do you know much about the lint error?

```
doc/conf.py:8:0: E0611: No name 'sorting' in module 'sphinx_gallery' (no-name-in-module)
```

I can't reproduce it locally, but will update my dependencies and see what's up with that.

**Comment by seisman** [comment]
Can't reproduce it with pylint 2.5.3, but CI uses pylint 2.6.0.

**Comment by weiji14** [comment]
> Can't reproduce it with pylint 2.5.3, but CI uses pylint 2.6.0.

Nope, passes locally for me on `pylint 2.6.0` too. Couldn't find anything relevant in the changelog at http://pylint.pycqa.org/en/latest/whatsnew/changelog.html#what-s-new-in-pylint-2-6-0 since we upgraded to pylint 2.5 a few months ago at #484

**Review by weiji14 — APPROVED** [review]

**Comment by seisman** [comment]
When I try to import an non-existent module like `from sphinx_gallery.foobar import FileNameSortKey`, I get the same pylint  error. However, sphinx_gallery is not installed in the Style Checking CI. Perhaps it's simply because sphinx_gallery is not installed.

**Pull request merged**